lib/auth.js

```diff
@@ -11,9 +10,24 @@ const ghauth = util.promisify(ghauthBase);
 
 export default lazy(auth);
 
+function errorExit(message) {
+  process.stderr.write(`${message}\n`);
+  process.exit(1);
+}
+
 function check(username, token) {
-  assert(typeof username === 'string' && /^[a-zA-Z0-9]*/.test(username));
-  assert(typeof token === 'string' && /^[0-9a-f]*/.test(token));
+  if (typeof username !== 'string') {
+    errorExit(`username must be a string, received ${typeof username}`);
+  }
+  if (!/^[a-zA-Z0-9-]+$/.test(username)) {
+    errorExit(`username must be alphanumeric, received ${username}`);
```

Comment:
Something more in line with GitHub error text (without the bits we're not actually checking for)?
```suggestion
    errorExit(`username may only contain alphanumeric characters or hyphens, received ${username}`);
```